docs/models_and_schedulers.rst

```diff
@@ -252,3 +252,146 @@ the following happens:
   requests.
 
 .. image:: images/sequence_example2.png
+
+.. _section-ensemble-models:
+
+Ensemble Models
+---------------
+
+With respect to the inference server's schedulers, an *ensemble* model is not
+an actual model. Instead, it specifies the dataflow between models involved as
+:cpp:var:`Step
+<nvidia::inferenceserver::ModelEnsembling::Step>`. The
+scheduler collects the output tensors in each step, provides them as input
+tensors for other steps according to the specification. In spite of that, the
+ensemble model is still viewed as a single model from an external view.
+
+Ensemble models are intended to be used to encapsulate a procedure that involves
+multiple models, such as "data preprocessing -> inference -> data postprocessing".
+Using ensemble models for this purpose can avoid the overhead of transfering
+intermediate tensors and minimize the number of requests to be sent
+on the client side.
+
+Note that the ensemble models will inherit the characteristics of the models
+involved, so the meta-data in the request header must comply with the models
+within the ensemble. For instance, if one of the models is stateful model, then
+the inference request for the ensemble model should contain the information
+mentioned in the previous :ref:`section <section-stateful-models>`, which will
+be provided to the stateful model by the scheduler. This is a limitation of
+ensemble where the stateful model in the ensemble must share the same flag.
+
+The :ref:`ensemble scheduler <section-ensemble-scheduler>` must be used for
```

Comment:
This should be moved up to lead the second paragraph.   "... regardless of the scheduler used by the models within the ensemble"